library/Erfurt/Store/Adapter/Stardog.php

```diff
@@ -0,0 +1,726 @@
+ * but use config.ini-dist-stardog as config.ini and adopt to your requirements
+ *
+ * To run stardog integration tests run: make test-integration-stardog
+ *
+ * Its tested with Stardog Version 4.0.1 (Docker available: https://github.com/Dockerizing/Stardog) and uses
+ * Guzzle 5.3 (https://github.com/guzzle/guzzle/tree/5.3) as HTTP Client
+ *
+ * @category Erfurt
+ * @package  Erfurt_Store_Adapter
+ * @author   Simeon Ackermann <amseon@web.de>
+ * @author   Norman Heino <norman.heino@gmail.com>
+ * @author   Philipp Frischmuth <pfrischmuth@googlemail.com>
+ */
+class Erfurt_Store_Adapter_Stardog implements Erfurt_Store_Adapter_Interface, Erfurt_Store_Sql_Interface    
+{
+    /**
+     * Prefix for blanknode identifiers.
+     * @var string
+     */    
+    const BLANKNODE_PREFIX = 'nodeID://';
+
+    /**
+     * The low-level client that is used to interact with the triple store.
+     *
+     * @var \Erfurt_Store_Adapter_Stardog_ApiClient
+     */
+    protected $_apiClient = null;
+
+    /**
+     * The inner SQL adapter.
+     *
+     * @var \Erfurt_Store_Sql_Interface
+     */
+    protected $_sqlAdapter = null;
+
+    /**
+     * The available graphs in this store.
+     * @var array
+     */
+    protected $_graphs = null;
+
+    /**
+     * Model imports cache.
+     * @var array
+     */
+    protected $_importedModels = array();
+    
+    private $_httpClientAdapter = null;
+    
+    /**
+     * Constructor.
+     *
+     * @throws Erfurt_Store_Adapter_Exception
+     */
+    public function __construct($adapterOptions = array())
+    {        
+    	// Access the connection in order to check whether it works.
+        $this->_apiClient = $this->connection($adapterOptions);
+
+        $dbs = $this->getDatabases();
+        if ( null === $dbs || ! in_array( $adapterOptions['database'], $dbs ) ) {
+            $this->createDatabase( $adapterOptions['database'] );
+        }
+        
+        // Create sql Zend Adapter if Versioning is enabled
+        $efApp = Erfurt_App::getInstance(false);
+        if ( $efApp->getVersioning() !== false ) {
+            $this->_sqlAdapter = $this->sqlConnection();
+        }
+    }
+
+    /**
+     * Destructor
+     *
+     * @throws Erfurt_Exception
+     */
+    public function __destruct()
+    {        
+        $this->_graphs = array();    
+        $this->_apiClient = null;
+        $this->_sqlAdapter = null;
+    }
+
+    /**
+     *  New Guzzle Http Client to Stardog
+     *
+     * @param array Stardog Options from config file
+     * @return GuzzleHttp/Client
+     */
+    public function connection(array $options = array())
+    {
+        return $this->_apiClient ? $this->_apiClient : new Erfurt_Store_Adapter_Stardog_ApiClient($options);
+    }
+
+    /**
+     * Connect new SQL Adapter for versioning
+     *
+     * @throws Erfurt_Store_Adapter_Exception if initialization failed
+     * @return \Erfurt_Store_Sql_Interface
+     */
+    public function sqlConnection()
```

Comment:
This implementation does not work with the test environment, since it uses a separate config file. Pleaser refer to a separate comment on the pull request, where I describe a possible solution